integration-tests/tests/tests/callbacks.rs

```diff
@@ -0,0 +1,174 @@
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert_eq!(function.params.len(), 0);
+    assert_eq!(function.callbacks.len(), 2);
+    let bool_schema = SchemaGenerator::default().subschema_for::<bool>();
+    let u32_schema = SchemaGenerator::default().subschema_for::<u32>();
+    assert_eq!(
+        function.callbacks[0],
+        AbiType {
+            type_schema: bool_schema,
+            serialization_type: AbiSerializationType::Json
+        }
+    );
+    assert_eq!(
+        function.callbacks[1],
+        AbiType {
+            type_schema: u32_schema,
+            serialization_type: AbiSerializationType::Json
+        }
+    );
+
+    Ok(())
+}
+
+#[test]
+#[named]
+fn test_callbacks_result() -> anyhow::Result<()> {
+    let abi_root = generate_abi_fn! {
+        pub fn foo(
+            &self,
+            #[callback_result] a: Result<String, near_sdk::PromiseError>,
+            #[callback_result] b: Result<u32, near_sdk::PromiseError>
+        ) {}
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert_eq!(function.params.len(), 0);
+    assert_eq!(function.callbacks.len(), 2);
+    let string_schema = SchemaGenerator::default().subschema_for::<String>();
+    let u32_schema = SchemaGenerator::default().subschema_for::<u32>();
+    assert_eq!(
+        function.callbacks[0],
+        AbiType {
+            type_schema: string_schema,
+            serialization_type: AbiSerializationType::Json
+        }
+    );
+    assert_eq!(
+        function.callbacks[1],
+        AbiType {
+            type_schema: u32_schema,
+            serialization_type: AbiSerializationType::Json
+        }
+    );
+
+    Ok(())
+}
+
+// FIXME: test does not work because callback_vec type is not unwrapped from `Vec<_>`.
+#[test]
+#[named]
+#[ignore]
+fn test_callbacks_vec() -> anyhow::Result<()> {
+    let abi_root = generate_abi_fn! {
+        pub fn foo(
+            &self,
+            #[callback_unwrap] a: bool,
+            #[callback_vec] b: Vec<u32>
+        ) {}
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert_eq!(function.params.len(), 0);
+    assert_eq!(function.callbacks.len(), 1);
+    let bool_schema = SchemaGenerator::default().subschema_for::<bool>();
+    let u32_schema = SchemaGenerator::default().subschema_for::<u32>();
+    assert_eq!(
+        function.callbacks[0],
+        AbiType {
+            type_schema: bool_schema,
+            serialization_type: AbiSerializationType::Json
+        }
+    );
+    assert_eq!(
+        function.callbacks_vec,
+        Some(AbiType {
+            type_schema: u32_schema,
+            serialization_type: AbiSerializationType::Json
+        })
+    );
+
+    Ok(())
+}
+
+// FIXME: test does not work because callback_vec type is not unwrapped from `Vec<_>`.
```

Comment:
updating the sdk commit and fixing this before bringing in or?

Reply:
Yeah since they are already approved I will just wait until the PRs are in